### PR TITLE
Refactor Golang-specific analysis to use the API

### PR DESCRIPTION
**Pull request by laomaiweng**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [X] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Use the API in `aalg`, as suggested in #3794, instead of executing `aF @@f:sym.go.*` directly.
No behavior change, so no doc/book update needed.

**Test plan**

Green CI.

**Closing issues**

N/A

### CODE REVIEW

File: librz/core/golang.c [hunk 1]
```diff
@@ -396,6 +396,24 @@ static ut32 core_recover_golang_functions_go_1_2(RzCore *core, GoPcLnTab *pclnta
...
 }
 
+static bool analyse_golang_symgo_function(RzFlagItem *fi, void *user) {
+	RzCore *core = (RzCore *)user;
+	rz_core_analysis_fcn(core, fi->offset, UT64_MAX, RZ_ANALYSIS_XREF_TYPE_NULL, 1);
+	return true;
+}
+
+/**
+ * \brief Analyse Golang symbols matching "syms.go.*"
```
Comment (XVilka) [inline librz/core/golang.c]: `sym.go` to match the actual logic?

**Review by XVilka — APPROVED** [review]
LGTM with minor comment problem

**Review by ret2libc — APPROVED** [review]
<3 Another great fix, thanks!

File: librz/core/golang.c (hunk 1 above)
Comment (laomaiweng) [reply librz/core/golang.c]: Fixed, thanks for spotting this!

**Pull request merged**